src/anomalib/utils/callbacks/metrics_manager.py

```diff
@@ -0,0 +1,173 @@
+        image_metrics (list[str] | None): List of image-level metrics.
+        pixel_metrics (list[str] | None): List of pixel-level metrics.
+    """
+
+    def __init__(
+        self,
+        task: TaskType = TaskType.SEGMENTATION,
+        image_metrics: list[str] | None = None,
+        pixel_metrics: list[str] | None = None,
+    ) -> None:
+        super().__init__()
+        self.task = task
+        self.image_metric_names = image_metrics
+        self.pixel_metric_names = pixel_metrics
+
+    def setup(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+        stage: str | None = None,
+    ) -> None:
+        """Setup image and pixel-level AnomalibMetricsCollection within Anomalib Model.
+
+        Args:
+            trainer (pl.Trainer): PyTorch Lightning Trainer
+            pl_module (AnomalyModule): Anomalib Model that inherits pl LightningModule.
+            stage (str | None, optional): fit, validate, test or predict. Defaults to None.
+        """
+        del trainer, stage  # These variables are not used.
+
+        image_metric_names = [] if self.image_metric_names is None else self.image_metric_names
+
+        pixel_metric_names: list[str]
+        if self.pixel_metric_names is None:
+            pixel_metric_names = []
+        elif self.task == TaskType.CLASSIFICATION:
+            pixel_metric_names = []
+            logger.warning(
+                "Cannot perform pixel-level evaluation when task type is classification. "
+                "Ignoring the following pixel-level metrics: %s",
+                self.pixel_metric_names,
+            )
+        else:
+            pixel_metric_names = self.pixel_metric_names
+
+        if isinstance(pl_module, AnomalyModule):
+            pl_module.image_metrics = create_metric_collection(image_metric_names, "image_")
+            pl_module.pixel_metrics = create_metric_collection(pixel_metric_names, "pixel_")
+
+            pl_module.image_metrics.set_threshold(pl_module.image_threshold.value)
+            pl_module.pixel_metrics.set_threshold(pl_module.pixel_threshold.value)
+
+    def on_validation_epoch_start(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+    ) -> None:
+        pl_module.image_metrics.reset()
+        pl_module.pixel_metrics.reset()
+
+    def on_validation_batch_end(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+        outputs: STEP_OUTPUT | None,
+        batch: Any,
+        batch_idx: int,
+        dataloader_idx: int = 0,
+    ) -> None:
+        if outputs is not None:
+            self._outputs_to_cpu(outputs)
+            self._collect_output(pl_module.image_metrics, pl_module.pixel_metrics, outputs)
+
+    def on_validation_epoch_end(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+    ) -> None:
+        self._update_metrics_threshold(pl_module)
```

Comment:
i remember you mentioned that in the previous pr, but don't exactly remember, so dropping again.

```suggestion
        self._update_threshold_metrics(pl_module)
```

Reply:
How about `_update_threshold`?